### PR TITLE
Refactor calls to `data_array_view`

### PR DESCRIPTION
**Pull request by galipremsagar**
## Description
This PR adapts to the breaking changes being introduced in: https://github.com/rapidsai/cudf/pull/12587

## Checklist
- [ ] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cuspatial/blob/HEAD/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.

### CODE REVIEW

File: python/cuspatial/cuspatial/utils/gis_utils.py [hunk 1]
```diff
@@ -35,5 +35,5 @@ def pip_bitmap_column_to_binary_array(polygon_bitmap_column, width):
     """Convert the bitmap output of point_in_polygon
     to an array of 0s and 1s.
     """
-    binary_maps = apply_binarize(polygon_bitmap_column.data_array_view, width)
+    binary_maps = apply_binarize(polygon_bitmap_column.data_array_view(mode="read"), width)
     return binary_maps
```
Comment (wence-) [inline python/cuspatial/cuspatial/utils/gis_utils.py:39]:
```
with acquire_spill_lock():
    return apply_binarize(...)
```

Or, I guess assign to the temporary and then return.

**Review by wence- — COMMENTED** [review]
Needs some spill lock acquisition I think.

File: ci/build_cpp.sh [hunk 2]
```diff
@@ -7,10 +7,14 @@ source rapids-env-update
 
 export CMAKE_GENERATOR=Ninja
 
+LIBCUDF_CHANNEL=$(rapids-get-artifact ci/cudf/pull-request/12587/046025a/cudf_conda_cpp_cuda11_$(arch).tar.gz)
```
Comment (galipremsagar) [inline ci/build_cpp.sh:10]: Temporary changes, will be reverted in few hours

**Review by ajschmidt8 — APPROVED** [review]

File: ci/build_python.sh [hunk 3]
```diff
@@ -12,12 +12,18 @@ rapids-print-env
 rapids-logger "Begin py build"
 
 CPP_CHANNEL=$(rapids-download-conda-from-s3 cpp)
+PY_VER=${RAPIDS_PY_VERSION//./}
```
Comment (galipremsagar) [inline ci/build_python.sh:15]: Temporary changes, will be reverted in few hours

File: ci/test_python.sh [hunk 4]
```diff
@@ -21,6 +21,9 @@ set -u
 rapids-logger "Downloading artifacts from previous jobs"
 CPP_CHANNEL=$(rapids-download-conda-from-s3 cpp)
 PYTHON_CHANNEL=$(rapids-download-conda-from-s3 python)
+PY_VER=${RAPIDS_PY_VERSION//./}
```
Comment (galipremsagar) [inline ci/test_python.sh:24]: Temporary changes, will be reverted in few hours

File: python/cuspatial/cuspatial/utils/gis_utils.py (hunk 1 above)
Comment (galipremsagar) [reply python/cuspatial/cuspatial/utils/gis_utils.py]: done 👍

File: python/cuspatial/cuspatial/core/spatial/distance.py [hunk 5]
```diff
@@ -91,7 +91,7 @@ def directed_hausdorff_distance(xs, ys, space_offsets):
         ys,
         as_column(space_offsets, dtype="uint32"),
     )
-    result = result.data_array_view
+    result = result.data_array_view()
     result = result.reshape(num_spaces, num_spaces)
     return DataFrame(result)
```
[the remark this answers is not in the document]
Comment (galipremsagar) [reply python/cuspatial/cuspatial/core/spatial/distance.py]: Done 👍

**Review by harrism — APPROVED** [review]

**Review by wence- — APPROVED** [review]

File: python/cuspatial/cuspatial/core/spatial/distance.py [hunk 6]
```diff
@@ -91,9 +91,10 @@ def directed_hausdorff_distance(xs, ys, space_offsets):
         ys,
         as_column(space_offsets, dtype="uint32"),
     )
-    result = result.data_array_view
-    result = result.reshape(num_spaces, num_spaces)
-    return DataFrame(result)
+    with cudf.core.buffer.acquire_spill_lock():
+        result = result.data_array_view(mode="write")
```
Comment (isVoid) [inline python/cuspatial/cuspatial/core/spatial/distance.py:95]: Why does this require `write` access to the view? I don't think the below changes the buffer.

**Review by isVoid — COMMENTED** [review]
Pending on a few temporary changes + 1 question.

File: python/cuspatial/cuspatial/core/spatial/distance.py (hunk 6 above)
Comment (galipremsagar) [reply python/cuspatial/cuspatial/core/spatial/distance.py:95]: Right, we are also performing a reshape too. Make sense to do a read-only view here.

Comment (galipremsagar) [inline python/cuspatial/cuspatial/core/spatial/distance.py:95]:
```suggestion
        result = result.data_array_view(mode="read")
```

**Review by isVoid — APPROVED** [review]
Thanks!

**Pull request merged**